libkbfs/bserver_remote.go

```diff
@@ -27,11 +26,18 @@ const (
 	BServerTokenExpireIn = 2 * 60 * 60 // 2 hours
 )
 
+type blockServerRemoteConfig interface {
+	diskBlockCacheGetter
+	codecGetter
+	signerGetter
+	currentInfoGetterGetter
+	logMaker
+}
+
 // BlockServerRemote implements the BlockServer interface and
 // represents a remote KBFS block server.
 type BlockServerRemote struct {
-	codec      kbfscodec.Codec
-	cig        currentInfoGetter
+	config     blockServerRemoteConfig
```

Comment:
I guess, to summarize my position, for small classes, having the constructor take all its parameters explicitly is best because it makes the dependencies explicit, and forces them to be instantiated before the class is. For large classes (with a lot of dependencies), a config struct also works, and seems to be common among the libraries I looked at (e.g., `crypto/tls`). Given a class that depended on `Config`, making a pared-down `fooConfig` interface is a pragmatic solution to avoid churn. I had some that were slightly different with an adapter type for `Config`, but I'm also okay with adding the new methods to `Config` (although it makes a big type even bigger). However, for new, small classes, and for existing classes with explicit dependencies, I view moving to a `fooConfig` type interface to be a step backwards, as it introduces undesired implicitness where there wasn't before, and makes possible a bunch of bugs (circular dependencies, uninstantiated dependencies).